pac4j-oidc/src/main/java/org/pac4j/oidc/credentials/extractor/OidcExtractor.java

```diff
@@ -76,7 +76,7 @@ public OidcCredentials extract(final WebContext context) throws HttpAction {
         if (state == null) {
             throw new TechnicalException("Missing state parameter");
         }
-        if (!state.equals(context.getSessionAttribute(OidcConfiguration.STATE_SESSION_ATTRIBUTE))) {
+        if (!state.equals(new State((String)context.getSessionAttribute(OidcConfiguration.STATE_SESSION_ATTRIBUTE)))) {
```

Comment:
What will happen if the value in the session is `null`? Does it work?

As we previously check that the `state` is not `null`, I'd prefer to have: `if (CommonHelper.areNotEquals(state.getValue(), context.getSessionAttribute(OidcConfiguration.STATE_SESSION_ATTRIBUTE)) {`